woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -1761,7 +1764,7 @@ def test_set_types(sample_df):
     pd.testing.assert_frame_equal(to_pandas(original_df), to_pandas(sample_df))
 
     sample_df.ww.set_types(logical_types={'is_registered': 'Integer'})
-    assert sample_df['is_registered'].dtype == 'Int64'
+    assert sample_df['is_registered'].dtype == 'int64'
```

Comment:
Thinking it might be useful to have a set of `set_types` checks that go from `Integer` to `IntegerNullable` and visa versa both when it's a valid conversion and when it's not. 

The stack trace for when it's not is this, and at first I was thinking that the final error we throw isn't super helpful for understanding why it's not allowed unless you have an in-depth understanding of pandas nullable dtypes, but I guess the `ValueError` that we catch is a part of the error, so maybe it's okay, but it might still be worth having an error message that further indicates it's a nullable dtype issue.

```python
~/.pyenv/versions/3.8.2/envs/woodwork/lib/python3.8/site-packages/pandas/core/arrays/masked.py in to_numpy(self, dtype, copy, na_value)
    219             ):
--> 220                 raise ValueError(
    221                     f"cannot convert to '{dtype}'-dtype NumPy array "

ValueError: cannot convert to 'int64'-dtype NumPy array with missing values. Specify an appropriate 'na_value' for this dtype.

During handling of the above exception, another exception occurred:

TypeConversionError                       Traceback (most recent call last)
<ipython-input-46-2489f63a00c9> in <module>
----> 1 sample_df.ww.init(logical_types={'age':'Integer'})
      2 display(sample_df.ww)
      3 sample_df.ww.set_types(logical_types={'is_registered':'Boolean'})

~/repositories/woodwork/woodwork/table_accessor.py in init(self, index, time_index, logical_types, make_index, already_sorted, schema, validate, use_standard_tags, **kwargs)
    139                 parsed_logical_types[name] = logical_type
    140 
--> 141                 updated_series = _update_column_dtype(series, logical_type)
    142                 if updated_series is not series:
    143                     self._dataframe[name] = updated_series

~/repositories/woodwork/woodwork/accessor_utils.py in _update_column_dtype(series, logical_type)
     87                 f'to type {new_dtype}. Please confirm the underlying data is consistent with ' \
     88                 f'logical type {logical_type}.'
---> 89             raise TypeConversionError(error_msg)
     90     return series
     91 

TypeConversionError: Error converting datatype for age from type Int64 to type int64. Please confirm the underlying data is consistent with logical type Integer.
```

Reply:
I added a test to test the valid and invalid conversions.

I understand your concern about the error message clarity. Since this is the same error that gets raised for any other invalid type conversion, I'm a little hesitant to add special logic to catch invalid conversion for the nullable/non-nullable dtypes and change the message based on that. I'm not sure it's worth the extra complexity in our error handling, but happy to discuss more if needed.

Reply:
yeah, as long as users are willing/able to look further into the stack trace, they'll see the pandas error thrown, so I probably agree that it's not worth the extra complexity